modules/neovim/init/basic.nix

```diff
@@ -158,112 +160,138 @@ in {
       default = "sensitive";
       description = "Set the case sensitivity of search";
     };
+
+    undoFile = {
+      enable = mkEnableOption "undofile for persistent undo";
+      path = mkOption {
+        type = either str luaInline;
+        default = mkLuaInline "vim.fn.stdpath('state') .. '/nvf/undo'";
+        defaultText = literalMD ''
+          ```nix
+          mkLuaInline "vim.fn.stdpath('state') .. '/nvf/undo'"
+          ```
+        '';
+        example = literalMD ''
+          ```nix
+          mkLuaInline "os.getenv('XDG_DATA_HOME') .. '/nvf/undo'"
+          ```
+        '';
+        description = "Path to the file in which undo history will be saved";
```

Comment:
Slightly misleading, this option's value gets passed along to `undodir`, which, as the name suggests, is a directory, in which there is an undofile for every file you make changes in.